flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -179,6 +179,8 @@ protected static void startClient(String[] args, Supplier<Terminal> terminalFact
         switch (mode) {
             case MODE_EMBEDDED:
                 final CliOptions options = CliOptionsParser.parseEmbeddedModeClient(modeArgs);
+                // check arg e.g -i -f
+                checkArgs(args);
```

Comment:
In the line above the args are already parsed into `CliOptions`. Wouldn't it be easier and better to do the validation on that object so we can avoid parsing the arguments (i.e. "-i", "-f") again in `checkArgs`?

Reply:
yeah , u are right , This is actually repeated  , I will check it in CliOptionsParser